github/messages.go

```diff
@@ -117,3 +120,41 @@ func validateSignature(signature string, payload, secretKey []byte) error {
 	}
 	return nil
 }
+
+// getWebhookEvent parses the payload and request, and build the Event.
+func getWebhookEvent(r *http.Request, payload []byte) (interface{}, error) {
+	eventType := r.Header.Get(eventTypeHeader)
+	if eventType == "push" {
+		pushEvent := WebHookPayload{}
+		if err := json.Unmarshal(payload, &pushEvent); err != nil {
+			return nil, err
+		}
+		return pushEvent, nil
+	}
+
+	event := Event{
+		Type:       &eventType,
+		RawPayload: (*json.RawMessage)(&payload),
+	}
+	return event.Payload(), nil
+}
+
+// ParseWebhook parses the request, validates the signature, and returns
+// an Event or a WebHookPayload.
+// secretKey is the GitHub Webhook secret message.
+//
+// Example usage:
+//
+//     func (s *GitHubEventMonitor) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+//       event, err := github.ParseWebHook(r, s.webhookSecretKey)
+//       if err != nil { ... }
+//       // Process event ...
+//     }
+//
+func ParseWebHook(r *http.Request, secretKey []byte) (interface{}, error) {
```

Comment:
This returns an `interface{}`. Can you please help me understand what one would do with the return value of type `interface{}` from `ParseWebHook`?

The example just says "// Process event ...", but it's unclear to me how one would do that.